crates/astria-composer/src/searcher.rs

```diff
@@ -1,358 +1,72 @@
-            // FIXME(https://github.com/astriaorg/astria/issues/215): need to set and track
-            // nonces for an actual fixed funded key/account.
-            // For now, each transaction is transmitted from a new account with nonce 0
-            let sequencer_key = ed25519_consensus::SigningKey::new(rand::thread_rng());
-            let nonce = Nonce::from(0);
-
-            // Pack into sequencer tx
-            let data = rollup_tx.rlp().to_vec();
-            let chain_id = chain_id.into_bytes();
-            let seq_action = SequencerAction::SequenceAction(SequenceAction::new(chain_id, data));
-            let unsigned_tx = UnsignedSequencerTx::new_with_actions(nonce, vec![seq_action]);
+    // FIXME: How do we start up multiple providers if we are reading from Config?
+    async fn setup_providers_from_config(
+        cfg: &Config,
+    ) -> Result<Vec<(Box<dyn StreamingClient<Error = eyre::Error>>, RollupChainId)>, eyre::Error>
+    {
+        let eth_client = EthProvider::connect(&cfg.execution_url)
+            .await
+            .wrap_err("failed connecting to eth")?;
 
-            // Sign transaction
-            Ok(unsigned_tx.into_signed(&sequencer_key))
-        });
+        Ok(vec![(Box::new(eth_client), "1".to_string())])
     }
 
-    fn handle_signed_tx(&mut self, tx: SignedSequencerTx) {
-        use sequencer_client::SequencerClientExt as _;
-        let client = self.sequencer_client.inner.clone();
-        self.submission_tasks.spawn(async move {
-            let rsp = client
-                .submit_transaction_sync(tx)
-                .await
-                .wrap_err("failed to submit transaction to sequencer")?;
-            if rsp.code.is_err() {
-                bail!(
-                    "submitting transaction to sequencer returned with error code; code: \
-                     `{code}`; log: `{log}`; hash: `{hash}`",
-                    code = rsp.code.value(),
-                    log = rsp.log,
-                    hash = rsp.hash,
-                );
-            }
-            Ok(())
-        });
-    }
+    /// Constructs a new Searcher service from config.
+    pub(super) async fn from_config(cfg: &Config) -> Result<Self, eyre::Error> {
+        let (collector, collector_recv_channel) = Collector::new();
 
-    /// Runs the Searcher
-    pub(super) async fn run(mut self) -> eyre::Result<()> {
-        use ethers::providers::{
-            Middleware as _,
-            StreamExt as _,
-        };
-        let wait_for_eth = self.wait_for_eth(5, Duration::from_secs(5), 2.0);
-        let wait_for_seq = self.wait_for_sequencer(5, Duration::from_secs(5), 2.0);
-        match tokio::try_join!(wait_for_eth, wait_for_seq) {
-            Ok(((), ())) => {}
-            Err(err) => return Err(err).wrap_err("failed to start searcher"),
+        let mut status = SearcherStatus::default();
+        for (provider, chain_id) in Self::setup_providers_from_config(cfg).await? {
```

Comment:
let's stick to a single provider for now - moving to multiple providers should be a separate pr